packages/grafana-ui/src/utils/dataLinks.ts

```diff
@@ -18,3 +18,8 @@
     };
   });
 };
+
+export const isCompactUrl = (url: string) => {
+  const compactExploreUrlRegex = /\/explore\?.*&(left|right)=\[(.*\,){2,}(.*){1}\]/g;
+  return compactExploreUrlRegex.test(url);
```

Comment:
I am extremely confident that a URL will not contain multiple repetitions of `/explore?`

Reply:
I think it can be dismissed. Also - would be worth adding a unit test to cover a few different URLs? The component test covers one URL but in a unit test it might be easier to test multiple URLs?

Reply:
not as elegant, but would it be worth it to simply parse the url and check for `left` and/or `right` query params and just get rid of this in the first place?

i'm trying to test something like `/explore?left=["lol",{lol:1}]` and doesn't seem to match the regex 🤔

Reply:
@Elfo404 is that a valid compact URL? I was going off https://github.com/grafana/grafana/blob/main/public/app/core/utils/explore.ts#L175 which seems to indicate that there should be at least 3 elements of the array for range from, range to, and datasource. The segments, which is optional, come after. So the regex looks for 2 or more elements of the array with a comma after them, and one at the end that doesn't have a comma.

Reply:
> (...) I'm trying to test something like /explore?left=["lol",{lol:1}] and doesn't seem to match the regex
> (...) I think that is correct based on this (...)

Could we please have a unit test for `isCompactUrl` to better show what cases are covered?

Reply:
Added more tests.